### PR TITLE
<Ruby group capstone - Catalog of my things>

### PR DESCRIPTION
**Pull request by Bestbynature**
## Features included in this pull request are as follows

Team Members: @Bestbynature , @Amazinggracee , @dev89s 

### Logic

- [x] Each class from the UML class diagram is created in a separate .rb file.
- [x] All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes have been implemented.
- [x] All methods from the UML class diagram have been implemented

### User interface

- [x] A main.rb file that serves as the console app entry point was created.
- [x] The console app, at the start, presents the user with a list of options to perform.

- Let users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.

- [x] The following options are available:
- List all books
- List all music albums
- List all movies
- List of games
- List all genres
- List all labels 
- List all authors
- List all sources
- Add a book
- Add a music album
- Add a movie
- Add a game
- [x] All data is preserved by saving collections in .json files.

### Database
A schema.sql file with tables that will be analogical to the structure of the classes in the app was created.

### Project Documentation
[Ruby Catalog Project Video](https://drive.google.com/file/d/1lY6Jd2gPRt60R6WqqXXy5hA5nCCLmdrW/view?usp=sharing)

### CODE REVIEW

**Review by osamaashraf6 — CHANGES_REQUESTED** [review]
Hi @Bestbynature 😊, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
SoCloseAlmostGIF (3)

## Highlights🎉🎉

- [ ] schema.sql file exists in the repo.😊
- [ ] data is saved after quitting the app💯
- [ ] created a main.rb file that will serve as their console app entry point.🚀
- [ ] All tests passed👌


## Required Changes ♻️

- [ ] You did a great effort at recording the video presentation, but I see that your video exceeded the 5 minutes, according to the requirements the maximum minutes for recording the video is 5 minutes, so please try to do that and make sure that the maximum is 5 minutes💯😊

| Requiremnts | Your video| 
| --- | --- |
|    image  |   image   |

_Check the comments under the review._

## Optional suggestions
- N/A


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  FOREIGN KEY (item_id) REFERENCES Item(id)
+);
+
+-- Table: Game
+CREATE TABLE Game (
+  item_id INT PRIMARY KEY,
+  multiplayer BOOLEAN,
+  last_played_at DATE,
+  FOREIGN KEY (item_id) REFERENCES Item(id)
+);
```
[the remark this answers is not in the document]
Comment (Bestbynature) [reply schema.sql:77]: Thank you so much @osamaashraf6. The A new video with the correct length has been Submitted and indexes have been added in the schema file.

[the remark this answers is not in the document]
Comment (Bestbynature) [reply schema.sql:77]: The new video link has been placed in the original pull request description but for emphasis' sake, here it is again:

[Ruby Capstone Group Project](https://drive.google.com/file/d/1lY6Jd2gPRt60R6WqqXXy5hA5nCCLmdrW/view?usp=sharing)

**Review by osamaashraf6 — APPROVED** [review]
Hi @Bestbynature  😊, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
MinionsGIF (3)

## Highlights🎉🎉

 - [ ] schema.sql file exists in the repo.😊
- [ ] data is saved after quitting the app💯
- [ ] created a main.rb file that will serve as their console app entry point.🚀
- [ ] All tests passed👌


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**